generator.go

```diff
@@ -0,0 +1,48 @@
+package justgo
+
+import (
+	"embed"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/gophers-mx/just-go/config"
+	"github.com/gophers-mx/just-go/pkg"
+)
+
+func Run(assets embed.FS, projectName, projectType *string, generator pkg.Generator) {
+	checkFlags(projectName, projectType)
+	cfg := setConfig(projectName, projectType, assets)
+
+	pkg.CreateDirectory()
+	generator.Run(cfg)
+}
+
+func checkFlags(projectName, projectType *string) {
+	if *projectName == "" {
+		fmt.Println("name cannot be blank")
+		os.Exit(1)
+	}
+
+	if *projectType == "" {
+		fmt.Println("type cannot be blank")
+		os.Exit(1)
+	}
+
+	checkedType, ok := config.ValidTypes[strings.ToLower(*projectType)]
+	if !ok {
+		fmt.Println("type is not valid")
```

Comment:
Should we provide structured logging or this is enough for now? Maybe standard library `log` can do the trick.

Reply:
We could pass a Logger interface to the `ProjectConfig` or `Cfg` with a `Log` method. But I don't really see the point. This is very fast, if we use the standard logger it would see like this:

```shell
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
2009/11/10 23:00:00 Generating "README.md" from template
```

The date would not change. The `fmt.Println` is cleaner imo.

```shell
Generating "README.md" from template
Generating "README.md" from template
Generating "README.md" from template
Generating "README.md" from template
Generating "README.md" from template
Generating "README.md" from template
Generating "README.md" from template
```

What do you think?